Allow contents in git.includes

marsam opened the pull request:
scenario: I'd like to specify the contents of the included path:
```nix
      includes = [
        { condition = "gitdir:~/work/";
          contents = {
            user = {
              name = "John Doe";
              email = "user@example.org";
            };
          };
        }
      ];
```

rycee commented on modules/programs/git.nix:
```diff
@@ -188,9 +194,13 @@ in
-            (i: with i; ''
+            (i: with i;
+            let
+              path' = if (contents != {}) then (pkgs.writeText "contents" (generators.toINI {} contents)) else path;
```
Instead of assembling `path'` here you can use the same trick as in [`file-type.nix`](https://github.com/rycee/home-manager/blob/0590c2a4f64871d00487639220d04955194bc172/modules/lib/file-type.nix#L93-L98) to default the `path` of the submodule to the INI content.

marsam replied:
thanks for pointing this out! this was new to me :)

rycee commented:
This is a good idea, thanks for implementing it! 🙂

rycee commented:
Oh, also. Please consider expanding the main Git test file: https://github.com/rycee/home-manager/blob/master/tests/modules/programs/git.nix

marsam commented:
Updated, and added a testcase
```
[marsam@meguca home-manager]$ nix-shell -A tests.run.all
bash: bind: warning: line editing not enabled
bash: bind: warning: line editing not enabled
bash: bind: warning: line editing not enabled
bash: bind: warning: line editing not enabled
bash: bind: warning: line editing not enabled
bash: bind: warning: line editing not enabled
files-executable: OK
files-hidden-source: OK
files-source-with-spaces: OK
files-text: OK
git-with-email: OK
git-with-most-options: OK
git-with-str-extra-config: OK
mbsync: OK
texlive-minimal: OK
tmux-emacs-with-plugins: OK
tmux-not-enabled: OK
tmux-vi-all-true: OK
xresources: OK
[marsam@meguca home-manager]$ 
```

rycee commented:
Great! Rebased to master in 6da88339f53f72eed62dedfbfb2e33acd0aa16da.
